Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -0,0 +1,801 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.UI;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Serialization;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// This script allows for an object to be movable, scalable, and rotatable with one or two hands. 
+    /// You may also configure the script on only enable certain manipulations. The script works with 
+    /// both HoloLens' gesture input and immersive headset's motion controller input.
+    /// </summary>
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/README_ManipulationHandler.html")]
+    public class ObjectManipulator : MonoBehaviour, IMixedRealityPointerHandler, IMixedRealityFocusChangedHandler
+    {
+        #region Public Enums
+
+        /// <summary>
+        /// Describes what pivot the manipulated object will rotate about when
+        /// you rotate your hand. This is not a description of any limits or 
+        /// additional rotation logic. If no other factors (such as constraints)
+        /// are involved, rotating your hand by an amount should rotate the object
+        /// by the same amount.
+        /// For example a possible future value here is RotateAboutUserDefinedPoint
+        /// where the user could specify a pivot that the object is to rotate
+        /// around.
+        /// An example of a value that should not be found here is MaintainRotationToUser
+        /// as this restricts rotation of the object when we rotate the hand.
+        /// </summary>
+        public enum RotateInOneHandType
```

Comment:
Can these public enums be moved [into their own files](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#public-classes-structs-and-enums-should-all-go-in-their-own-files)?